src/main/golo/assertions.golo

```diff
@@ -0,0 +1,293 @@
+  let start_root = java.io.File(path)
+  let files = start_root: listFiles(): asList()
+
+  files: each(|file|{
+    if file: isDirectory() is true {
+      filesDiscover(file: getAbsolutePath(), extension, listOfFiles, root)
+    } else {
+      if file: getAbsoluteFile(): getName(): endsWith("." + extension) is true {
+        var fileName = file: getAbsolutePath(): toString(): split(root): get(1)
+        listOfFiles: add(fileName)
+      }
+    }
+  })
+  return listOfFiles
+}
+# this is the list of golo files in a project
+let goloFiles = list[]
+
+struct testsReport = {
+  passed, failed
+}
+# instance of testsReport to keep results of assertions
+let report = testsReport(0,0)
+----
+Display results of assertions
+----
+function displayTestsReport = {
+  println("tests: passed:"+report: passed()+" failed:"+report: failed())
+}
+----
+get the results of assertions
+
+- *return* results of assertions (struct testsReport)
+----
+function getTestsReport = -> report
+
+# We need to know if we can use ansi codes
+let not_windows = likelySupported()
+
+----
+Helper to decompose a stacktrace line about error with a Golo script
+
+- *param* `path` the path to parse (String)
+----
+local function getDetailsOfGoloError = |stackTraceLine| {
+  let split = stackTraceLine: toString(): split("\\(")
+  let moduleNameAndMethod = split: get(0)
+  let fileName = split: get(1): split(":"): get(0)
+  let lineNumber = Integer.valueOf(split: get(1): split(":"): get(1): split("\\)"): get(0))
+
+  if(not_windows) { fg_blue() }
+
+  if(goloFiles: size(): equals(0)) {
+    filesDiscover(currentDir(), "golo", goloFiles, currentDir())
+  }
+
+  goloFiles: filter(|filePathName| {
+    return filePathName: endsWith(fileName)
+  }): each(|filePathName| {
+    let lines = java.nio.file.Files.readAllLines(
+      java.nio.file.Paths.get(currentDir()+filePathName),
+      java.nio.charset.Charset.forName("UTF-8")
+    )
+    let moduleNameInSourceCode = lines:get(0): split("module "): get(1): trim()
+    if(moduleNameAndMethod: startsWith(moduleNameInSourceCode+".")) {
+      println("  line number: " + lineNumber + ": " + lines: get(lineNumber - 1) + " ...")
+    }
+  })
+}
+
+----
+Helper to test predicate
+
+- *param* `predicate` this is a closure (FunctionReference)
+- *param* `allStackTrace` if true, the complete Java StackTrace is displayed, if false, only Golo stuff is displayed (Boolean)
+- *param* `getSuccessMessage` this is a closure that return a string (FunctionReference)
+- *param* `onSuccess` this is the callBack if success (FunctionReference)
+- *param* `getErrorMessage` this is a closure that return a string (FunctionReference)
+- *param* `exitWhenFailed` if true, then the program aborts at the first error (Boolean)
+- *param* `onError` this is the callBack if error (FunctionReference)
+----
+function assert = |predicate, allStackTrace, getSuccessMessage, onSuccess, getErrorMessage, exitWhenFailed, onError| {
+
```

Comment:
@yloiseau the assertions module allows you to write your own assert methods like [this `assertEquals`](https://github.com/k33g/golo-lang/commit/d0e897b65ed06cc7cd7dc48417e89f8a7f292f5f#diff-72c06ea103c6fb38c87a40f873511f36R252)

so we can add it easily